x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/handle_api_errors.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HttpResponse } from 'src/core/public';
+
+import { FlashMessagesLogic, IFlashMessage } from './';
+
+/**
+ * The API errors we are handling can come from one of two ways:
+ *  - When our http calls recieve a response containing an error code, such as a 404 or 500
+ *  - Our own JS while handling a successful response
+ *
+ * In the first case, if it is a purposeful error (like a 404) we will receive an
+ * `errors` property in the response's data, which will contain messages we can
+ * display to the user.
+ */
+interface IErrorResponse {
+  statusCode: number;
+  error: string;
+  message: string;
+  attributes: {
+    errors: string[];
+  };
+}
+
+/**
+ * Converts API/HTTP errors into user-facing Flash Messages
+ */
+export const handleAPIError = (error: HttpResponse<IErrorResponse>) => {
+  const defaultErrorMessage = 'An unexpected error occurred';
+
+  const errorFlashMessages: IFlashMessage[] = Array.isArray(error?.body?.attributes?.errors)
+    ? error.body!.attributes.errors.map((message) => ({ type: 'error', message }))
+    : [{ type: 'error', message: defaultErrorMessage }];
+
+  FlashMessagesLogic.actions.setFlashMessages(errorFlashMessages);
```

Comment:
I was actually pretty blown away that this Just Worked(™️) within **both** a Logic file:

https://github.com/elastic/kibana/blob/063882344769eb8e606e6b5d2d1e30e5b3264255/x-pack/plugins/enterprise_search/public/applications/workplace_search/views/overview/overview_logic.ts#L131-L138

and a React component:

https://github.com/elastic/kibana/blob/063882344769eb8e606e6b5d2d1e30e5b3264255/x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx#L52-L61

Kea is amazing! 💞